modules/dashboard/widgets/xy-ratio/main.tf

```diff
@@ -0,0 +1,57 @@
+variable "title" { type = string }
+variable "legend" { default = "" }
+variable "numerator_group_by_fields" { default = [] }
+variable "denominator_group_by_fields" { default = [] }
+variable "numerator_filter" { type = list(string) }
+variable "denominator_filter" { type = list(string) }
+variable "plot_type" { default = "LINE" }
+variable "alignment_period" { default = "60s" }
+variable "numerator_align" { default = "ALIGN_RATE" }
+variable "numerator_reduce" { default = "REDUCE_SUM" }
+variable "denominator_align" { default = "ALIGN_RATE" }
+variable "denominator_reduce" { default = "REDUCE_SUM" }
+variable "thresholds" { default = [] }
+
+// https://cloud.google.com/monitoring/api/ref_v3/rest/v1/projects.dashboards#XyChart
+output "widget" {
+  value = {
+    title = var.title
+    xyChart = {
+      chartOptions = { mode = "COLOR" }
+      dataSets = [{
+        minAlignmentPeriod = var.alignment_period
+        plotType           = var.plot_type
+        targetAxis         = "Y1"
+        legendTemplate     = var.legend
+        timeSeriesQuery = {
+          // https://cloud.google.com/monitoring/api/ref_v3/rest/v1/projects.dashboards#timeseriesfilterratio
+          timeSeriesFilterRatio = {
+            numerator = {
+              filter = join("\n", var.numerator_filter)
+              aggregation = {
+                alignmentPeriod    = var.alignment_period
+                perSeriesAligner   = var.numerator_align
+                crossSeriesReducer = var.numerator_reduce
+                groupByFields      = var.numerator_group_by_fields
```

Comment:
struggling between letting users choose their own poison [full control], or handle the common case where numerator and denominator has be configured so it aligns and can be divided properly, using the same metric, and just filtered differently

Reply:
yeah I feel the same way. On one hand, a ratio could really be any two metrics, but in reality how often will they not have the same baseline.

What about both? I could add a second widget that makes the assumption the numerator is just a more filtered version of the denominator metric to simplify the common case, but leave `xy-ratio` as is in case someone wants full control.